src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java

```diff
@@ -556,6 +608,99 @@ private void emitMetricsForStorageBucket(boolean succeeded, BatchSinkContext con
     }
   }
 
+  /**
+   * Configures metadata update in Dataplex if the schema is user managed
+   *
+   * @param credentials
+   * @param assetFullPath
+   * @param storageSystem
+   * @param schema
+   * @throws IOException
+   */
+  private void prepareMetadataUpdateWithUserManagedSchema (
+    GoogleCredentials credentials, String assetFullPath, StorageSystem storageSystem, Schema schema
+  ) throws IOException {
+    String entityID = config.getTable().replaceAll("[^a-zA-Z0-9_]", "_");
+    MetadataServiceClient metadataServiceClient =
+      DataplexUtil.getMetadataServiceClient(credentials);
+    try {
+      entityBean =
+        metadataServiceClient.getEntity(GetEntityRequest.newBuilder().setName(EntityName.of(
+            config.tryGetProject(), config.getLocation(), config.getLake(), config.getZone(), entityID).toString())
+                                          .setView(GetEntityRequest.EntityView.FULL)
+                                          .build());
+    } catch (ApiException e) {
+      // ignore
```

Comment:
why are we ignoring APIException?

Reply:
Because if entityBean is null (which gives an ApiException (404)), we'd create one, if it exists, we update it.

Reply:
I see. should we ignore the exception only if e.getCode() == 404 (add failure to collector for any other errors)?